cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -433,9 +433,13 @@ public void createAppDetailsArchive(ZipOutputStream zipOut) {
    * @return a collection of verion strings
    */
   public Collection<String> getAppVersions(NamespaceId namespaceId, String application) {
-    return store.getAllAppVersionsAppIds(namespaceId.app(application)).stream()
-      .map(ApplicationId::getVersion)
-      .collect(Collectors.toList());
+    List<String> appVersions = store.getAllAppVersionsAppIds(namespaceId.app(application)).stream()
+      .map(ApplicationId::getVersion).collect(Collectors.toList());
+    // Hide -SNAPSHOT when more than one version of the app exists
+    if (appVersions.size() > 1 && appVersions.contains(ApplicationId.DEFAULT_VERSION)) {
```

Comment:
but the filtering must happen on `appVersions.size() > 1` condition too right?

Reply:
right, so something like:

```
    Collection<ApplicationId> ids = store.getAllAppVersionsAppIds(namespaceId.app(application));
    if (ids.size() > 1) {
      return ids.stream()
        .filter(id -> !ApplicationId.DEFAULT_VERSION.equals(id.getVersion()))
        .map(ApplicationId::getVersion)
        .collect(Collectors.toList());
    }
    return ids.stream()
      .map(ApplicationId::getVersion)
      .collect(Collectors.toList());
```

Reply:
In general it's also not good to have these Store methods that return everything in a collection, as things can go out of memory. Though since these are just ids it should be ok.

Reply:
okay, got it. Thanks!